packages/formation-react/src/components/modal/Modal.jsx

```diff
@@ -25,7 +25,9 @@ class Modal extends React.Component {
 
   componentWillUnmount() {
     if (this.props.visible) {
-      this.teardownModal();
+      setTimeout(() => {
+        this.teardownModal()
+      }, 0);
     }
```

Comment:
This will hold up closing the modal / halting the teardown until after teardown is complete.